test-it/karma/publicapi/FPSync.Test.js

```diff
@@ -0,0 +1,67 @@
+describe('The Vireo CoreHelpers setFPSyncFunction api', function () {
+    'use strict';
+    // Reference aliases
+    var Vireo = window.NationalInstruments.Vireo.Vireo;
+    var vireoRunner = window.testHelpers.vireoRunner;
+    var fixtures = window.testHelpers.fixtures;
+
+    var vireo;
+
+    var publicApiFPSyncSimpleViaUrl = fixtures.convertToAbsoluteFromFixturesDir('publicapi/FPSyncSimple.via');
+    var publicApiFPSyncUtf8ViaUrl = fixtures.convertToAbsoluteFromFixturesDir('publicapi/FPSyncUtf8.via');
+
+    beforeAll(function (done) {
+        fixtures.preloadAbsoluteUrls([
+            publicApiFPSyncSimpleViaUrl,
+            publicApiFPSyncUtf8ViaUrl
+        ], done);
+    });
+
+    beforeEach(function () {
+        vireo = new Vireo();
+    });
+
+    it('can perform a simple fpsync', function (done) {
+        var tracker = jasmine.createSpy();
+        vireo.coreHelpers.setFPSyncFunction(tracker);
+        var runSlicesAsync = vireoRunner.rebootAndLoadVia(vireo, publicApiFPSyncSimpleViaUrl);
+
+        runSlicesAsync(function (rawPrint, rawPrintError) {
+            expect(rawPrint).toBeEmptyString();
+            expect(rawPrintError).toBeEmptyString();
+            expect(tracker).toHaveBeenCalled();
+            expect(tracker).toHaveBeenCalledWith('Hello World');
+
+            done();
+        });
+    });
+
+    it('can perform a fpsync with UTF8 characters', function (done) {
+        var viName = 'MyVI';
+
+        var trackerCalls = [];
+        var tracker = function (fpSyncString) {
+            var myDouble = vireo.eggShell.readDouble(viName, 'myDouble');
+            trackerCalls.push({
+                myDouble: myDouble,
+                fpSyncString: fpSyncString
+            });
+        };
+
+        vireo.coreHelpers.setFPSyncFunction(tracker);
+        var runSlicesAsync = vireoRunner.rebootAndLoadVia(vireo, publicApiFPSyncUtf8ViaUrl);
+
+        runSlicesAsync(function (rawPrint, rawPrintError) {
+            var fpSyncString = 'Iñtërnâtiônàlizætiøn\0☃💩';
```

Comment:
I wonder whether we should decode this expected string from UTF-8 encoded byte string (in hex) of the actual string in the .via file .